langchain/src/document_loaders/fs/unstructured.ts

```diff
@@ -95,26 +143,34 @@ export class UnstructuredLoader extends BaseDocumentLoader {
 
 export class UnstructuredDirectoryLoader extends DirectoryLoader {
   constructor(
-    public webPath: string,
-    public directoryPath: string,
+    directoryPathOrLegacyApiUrl: string,
+    optionsOrLegacyDirectoryPath: UnstructuredOptions | string,
     public recursive: boolean = true,
```

Comment:
I think ideally we move these into the options arg too? @jacoblee93

Reply:
Done, left them that way originally to mirror the underlying `DirectoryLoader` syntax but agree that this is nicer.